### PR TITLE
Add render props to <Field />

### PR DESCRIPTION
**Pull request by julienvincent**
This PR allows the usage pattern described in #182

### CODE REVIEW

**Comment by jaredpalmer** [comment]
Is there really a need for this though? Just pass props if you need to write this inline.

**Comment by julienvincent** [comment]
Passing props as apposed to using the context api adds a significant amount of ceremony, especially in complex forms spread across multiple nested components.

Using  the `<Field />` component api is very elegant, but unfortunately it does not conform to the normal React `onChange` + `value` props pattern. This means that integration with existing custom components is not as simple as just passing them to the `component` prop as they won't be compatible.

#### tl;dr
Adding support for inline lambdas will make compatibility with various custom components much simpler and will allow smoother and easier migration to Formik. Less tears :)

**Comment by jaredpalmer** [comment]
By passing props i meant something like this. (this is the way it worked before context 0.9) 

```js
<Formik
   render={props => 
    <form onSubmit={props.handleSubmit}>
      <input 
         name="email"
         type="email"
         onChange={props.handleChange} 
         onBlur={props.handleBlur} 
         value={values.email} 
       />
    </form>}
/>

```

**Comment by julienvincent** [comment]
Yea I'm with you on that. My use case is one in which my form input components are sitting in several nested components (complex capture forms) and so those Formik props would need to travel down through two or perhaps three container components before arriving at their respective input component.

Using the context api cleans this up significantly (at the cost perhaps of obscuring links between data and capture).

**Comment by jaredpalmer** [comment]
I found myself wanting this the other day. Can you add render, component, and children ? should be similar to logic found in <Formik />

**Comment by julienvincent** [comment]
You didn't finish writing the comment but I'm assuming you are referring to react-router? If so it might make sense to leave out the `children` prop according to [this react-router issue](https://github.com/ReactTraining/react-router/issues/5583)

**Comment by jaredpalmer** [comment]
forgot to escape a character

**Comment by julienvincent** [comment]
Ok I see now. I will make the changes tomorrow 👍

**Comment by jaredpalmer** [comment]
Yeah just implement children for now to stay consistent. Just one line. Once React Router decides on children we'll follow suit.

**Comment by julienvincent** [comment]
Alright I added a `render` and `children` as props. The implementation prioritizes render => children => component, not sure if you would like to change these priorities?

**Comment by julienvincent** [comment]
Any updates here?

**Comment by jaredpalmer** [comment]
Gonna merge and make the changes after.
